server/src/test/java/org/elasticsearch/cluster/metadata/DataStreamTests.java

```diff
@@ -60,7 +61,57 @@ protected DataStream createTestInstance() {
 
     @Override
     protected DataStream mutateInstance(DataStream instance) {
-        return null;// TODO implement https://github.com/elastic/elasticsearch/issues/25929
+        var name = instance.getName();
+        var indices = instance.getIndices();
+        var generation = instance.getGeneration();
+        var metadata = instance.getMetadata();
+        var isHidden = instance.isHidden();
+        var isReplicated = instance.isReplicated();
+        var isSystem = instance.isSystem();
+        var allowsCustomRouting = instance.isAllowCustomRouting();
+        var indexMode = instance.getIndexMode();
+        var lifecycle = instance.getLifecycle();
+        switch (between(0, 9)) {
+            case 0 -> name = randomAlphaOfLength(10);
+            case 1 -> indices = DataStreamTestHelper.randomIndexInstances();
+            case 2 -> generation = instance.getGeneration() + randomIntBetween(1, 10);
+            case 3 -> metadata = randomBoolean() && metadata != null ? null : Map.of("key", randomAlphaOfLength(10));
+            case 4 -> {
+                if (isHidden) {
+                    isHidden = false;
+                    isSystem = false; // A system data stream must always be hidden
```

Comment:
This comment should be at the line above I think

Reply:
This was intentional, but maybe I should change the content. What I want to say is that here we mutate the `isHidden`, however, we need to ensure that it's not a system index as well, otherwise we will generate an invalid combination. I will fix it.